julee_example/use_cases/extract_assemble_data.py

```diff
@@ -291,14 +298,42 @@ async def _retrieve_all_queries(
             assembly_specification.knowledge_service_queries.values()
         )
 
-        # Use batch retrieval for efficiency
-        query_results = await self.knowledge_service_query_repo.get_many(
-            query_ids
-        )
-
+        # TODO: TEMPORAL SERIALIZATION ISSUE - Replace with get_many when
+        # fixed
+        #
+        # Issue: Complex return type
+        # Dict[str, Optional[KnowledgeServiceQuery]] from get_many causes
+        # Temporal's type system to fall back to typing.Any, resulting in
+        # Pydantic models being deserialized as plain dictionaries instead of
+        # model instances.
+        #
+        # Error: "SERIALIZATION ISSUE DETECTED: parameter
+        # 'queries'['query-id'] is dict instead of KnowledgeServiceQuery!"
+        #
+        # Root Cause: Temporal's type resolution cannot handle the complex
+        # nested generic type Dict[str, Optional[T]] and passes typing.Any to
+        # the data converter, which then deserializes to plain dicts.
+        #
+        # Investigation: Full analysis showed:
+        # - Data converter debug output confirming typing.Any fallback
+        # - Repository type resolution working correctly
+        # - Guard check system detecting the exact issue
+        # - Evidence that simpler types (Optional[T]) work fine
+        #
+        # Temporary Fix: Use individual get() calls which return Optional[T]
+        # that Temporal handles correctly.
+        #
+        # Future Solutions:
+        # 1. Fix Temporal's type resolution for complex nested generics
+        # 2. Create custom data converter for this specific type pattern
+        # 3. Simplify repository interface to avoid Optional in batch
+        #    operations
+        #
+        # Currently using individual get calls to avoid complex type
+        # serialization issue
```

Comment:
Oh, that's a bummer. Maybe we need to push a fix upstream, after discussing with Temporal community if this is a bug or if they thing we are abusing something?

Reply:
Possibly - though I'd need to spend some more time testing with a smallest reproducible error. Most likely cause is going to be something I'm not doing correctly, but keen to find out.